msal/application.py

```diff
@@ -50,6 +50,17 @@ def decorate_scope(
     return list(decorated)
 
 
+def extract_certs(public_cert_content):
+    # Parses raw public certificate file contents and returns a list of strings
+    # Usage: headers = {"x5c": extract_certs(open("my_cert.pem").read())}
+    public_certificates = re.findall(
+        r'\-+BEGIN CERTIFICATE.+\-+(?P<cert_value>[^-]+)\-+END CERTIFICATE.+\-+',
```

Comment:
Based on the [PEM format definition](https://tools.ietf.org/html/rfc7468#section-2), it seems the two `.+` in existing regex are redundant. Also we can reduce the use of `\-+` regex magic, in favor of plain text.

Will this one be slightly more readable? `r"(?:-----BEGIN CERTIFICATE-----)(?P<cert_value>[^-]+)(?:-----END CERTIFICATE-----)"`
By the way, there are plenty of [online regex playground](https://regex101.com/) for quick prototyping.

Reply:
Yes ! I did use that to try out different things. I can make it more readable by doing that just a little differently. Will address in the next commit. Stay tuned :)